hawc/apps/assessment/views.py

```diff
@@ -351,25 +356,34 @@ def get_success_url(self):
 
 
 # Dataset views
-class DatasetCreate(BaseCreate):
+class DatasetPublishedMixin:
+    def get_object(self, **kwargs):
+        obj = super().get_object(**kwargs)
+        if obj.assessment.user_can_edit_object(self.request.user) or obj.published:
```

Comment:
this if statement is used in a few places. Should we move it to the dataset instance?

```python
def user can view(self, user):
    return self.assessment.user_can_edit_object(user) or self.published

...

if not obj.user_can_view(self.request.user):
    raise PermissionDenied()
```

Reply:
I ended up doing 
```
def user_can_view(self, user) -> bool:
        return (
            self.published
            and self.assessment.user_can_view_object(user)
            or self.assessment.user_can_edit_object(user)
        )
```
There is some redundancy with the `user_can_view_object` check and the assessment level permissions, but I figure thats more correct than this method relying on those base permissions already being in place.

Reply:
nice job